monitor/dockermonitor/docker.go

```diff
@@ -400,13 +404,16 @@ func (d *dockerMonitor) startDockerContainer(dockerInfo *types.ContainerJSON) er
 	errorChan := d.puHandler.HandlePUEvent(contextID, monitor.EventStart)
 
 	if err := <-errorChan; err != nil {
-		if err := d.dockerClient.ContainerStop(context.Background(), dockerInfo.ID, &timeout); err != nil {
-			log.WithFields(log.Fields{
-				"package": "monitor",
-				"error":   err.Error(),
-			}).Warn("Failed to stop bad container")
+		if d.killContainerError {
+			if err := d.dockerClient.ContainerStop(context.Background(), dockerInfo.ID, &timeout); err != nil {
+				log.WithFields(log.Fields{
+					"package": "monitor",
+					"error":   err.Error(),
+				}).Warn("Failed to stop bad container")
+			}
+			return fmt.Errorf("Policy cound't be set - container was killed")
 		}
-		return fmt.Errorf("Policy cound't be set - container was killed")
+		return fmt.Errorf("Policy cound't be set - container was kept alive per policy")
```

Comment:
If we keep the container alive why are we returning error ? what will happen to this container as we move forward. I think we can get into weird states.